### PR TITLE
Fix a couple of common PHP issues

### PR DESCRIPTION
**Pull request by magicsunday**
I did some common rework of the PHP classes

### CODE REVIEW

File: Classes/Controller/SelectImageController.php [hunk 1]
```diff
@@ -1,65 +1,87 @@
...
-            if (!$bparams[3]) {
+
+            if (isset($bparams[3]) && ($bparams[3] === '')) {
                 $bparams[3] = $GLOBALS['TYPO3_CONF_VARS']['GFX']['imagefile_ext'];
                 $_GET['bparams'] = implode('|', $bparams);
             }
         }
-        $this->mode = 'file';
+
+        $this->mode = GeneralUtility::_GET('mode') ?: 'file';
```
Comment (aseemann) [inline Classes/Controller/SelectImageController.php:77]: I'm not really sure, may be I'm paranoid but this looks like an potential XSS Risk.

**Review by aseemann — COMMENTED** [review]
Hi Rico, 

looks good to me so far, besides one comment or place where the risk of cross site scripting isn't really clear to me. Please have a look on it.

File: Classes/Controller/SelectImageController.php (hunk 1 above)
Comment (magicsunday) [reply Classes/Controller/SelectImageController.php]: I removed the assignment in commit #07769e4. The value is passed through the route to the parent, no need to assign it again.

**Review by TechnicalAthere — APPROVED** [review]
-  Code Review passed
-  Functional test passed

**Pull request merged**